src/instructlab/model/train.py

```diff
@@ -7,15 +7,93 @@
 import shutil
 
 # Third Party
+from instructlab.training import TorchrunArgs, TrainingArgs, run_training
 import click
 import torch
 
 # First Party
 from instructlab import utils
 
 
+class TorchDeviceParam(click.ParamType):
+    """Parse and convert device string
+
+    Returns a torch.device object:
+    - type is one of 'cpu', 'cuda', 'hpu'
+    - index is None or device index (e.g. 0 for first GPU)
+    """
+
+    name = "deviceinfo"
+    supported_devices = {"cuda", "cpu", "hpu"}
+
+    def convert(self, value, param, ctx) -> "torch.device":
```

Comment:
I don't think this output annotation works, I haven't seen it before.

Reply:
it already exists, should we leave it for now?

Reply:
yeah if it's not breaking. we can refactor later.